net/epee-encoding/src/lib.rs

```diff
@@ -212,6 +212,87 @@ fn write_epee_value<T: EpeeValue, B: BufMut>(val: T, w: &mut B) -> Result<()> {
     val.write(w)
 }
 
+/// Write a byte array to `w` with [`write_varint`].
+///
+/// This function:
+/// - Writes the length of `t`'s bytes into `w` using [`write_varint`]
+/// - Writes `t`'s bytes into `w`
+///
+/// It is used as the internal [`EpeeValue::write`]
+/// implementation of byte-like containers such as:
+/// - [`EpeeValue::<Vec<u8>>::write`]
+/// - [`EpeeValue::<String>::write`]
+///
+/// # Errors
+/// This will error if:
+/// - [`write_varint`] fails
+/// - `w` does not have enough capacity
+///
+/// # Example
+/// ```rust
+/// let t: [u8; 8] = [3, 0, 0, 0, 1, 0, 0, 0];
+/// let mut w = vec![];
+///
+/// epee_encoding::write_bytes(t, &mut w).unwrap();
+///
+/// assert_eq!(w.len(), 9); // length of bytes + bytes
+/// assert_eq!(w[1..], t);
+/// ```
+pub fn write_bytes<T: AsRef<[u8]>, B: BufMut>(t: T, w: &mut B) -> Result<()> {
+    let bytes = t.as_ref();
+    let len = bytes.len();
+
+    write_varint(len.try_into()?, w)?;
+
+    if w.remaining_mut() < len {
+        return Err(Error::IO("Not enough capacity to write bytes"));
+    }
+
+    w.put_slice(bytes);
+
+    Ok(())
+}
+
+/// Write an [`Iterator`] of [`EpeeValue`]s to `w` with [`write_varint`].
+///
+/// This function:
+/// - Writes the length of `i`'s bytes into `w` using [`write_varint`]
```

Comment:
This should say the length of the iterator, not the length of the bytes, also is it supposed to be `i` or `iterator`?